src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,833 @@
+        }
+
+        (address lpToken,,, address crvRewards,,) = CONVEX_BOOSTER.poolInfo(_removePid);
+        IBaseRewardPool baseRewardPool = IBaseRewardPool(crvRewards);
+
+        uint256 stakedBal = baseRewardPool.balanceOf(address(this));
+        if (stakedBal > 0) {
+            revert CurveLpStillStaked(lpToken, crvRewards, stakedBal);
+        }
+
+        // NOTE: verify pending rewards and claim. Processing is done separately
+        if (baseRewardPool.earned(address(this)) > 0) {
+            baseRewardPool.getReward();
+        }
+
+        // NOTE: while removing the info from storage, we ensure that allowance is set back to zero
+        IERC20MetadataUpgradeable(lpToken).safeApprove(address(CONVEX_BOOSTER), 0);
+
+        // NOTE: indeed if nothing is staked, then remove from storage
+        pids.remove(_removePid);
+        assets.remove(lpToken);
+        baseRewardPools.remove(crvRewards);
+    }
+
+    /// @notice Sweep the full contract's balance for a given ERC-20 token. Can only be called by owner.
+    /// @param token The ERC-20 token which needs to be swept
+    function sweep(address token) external onlyOwner {
+        IERC20MetadataUpgradeable erc20Token = IERC20MetadataUpgradeable(token);
+        uint256 balance = erc20Token.balanceOf(address(this));
+        erc20Token.safeTransfer(msg.sender, balance);
+        emit ERC20Swept(token, balance);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Owner/Manager
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Claim and process CRV, CVX & FXS rewards, selling them for stables: USDC & FRAX
+    /// @dev This can be called by the owner or manager to opportunistically harvest in good market conditions.
+    /// @return processed_ An array containing addresses and amounts of harvested tokens (i.e. tokens that have finally
+    ///                    been swapped into).
+    function processRewards() external onlyOwnerOrManager returns (TokenAmount[] memory processed_) {
+        processed_ = _processRewards();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Keeper
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice A function to process pending CRV, CVX & FXS rewards at regular intervals. Can only be called by the
+    ///         keeper when the contract is not paused.
+    /// @param _performData The calldata to be passed to the upkeep function. Not used!
+    function performUpkeep(bytes calldata _performData) external override onlyKeeper whenNotPaused {
+        _processRewardsKeeper();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // INTERNAL
+    ////////////////////////////////////////////////////////////////////////////
+
+    function _createPrivateVault(uint256 _pid) internal {
```

Comment:
Since we could be calling this multiple times (we could create 2 vaults with same pid)

Is there ever a reason to remove the allowance we are granting?

`IERC20MetadataUpgradeable(stakingTokenAddr).safeApprove(vaultAddr, type(uint256).max);`

Reply:
in this particular case, do not see a concern leaving the allowance granted to the private vault, since concerning methods are permissioned to only `owner` (in this case the creator of the vault, the avatar) to trigger methods which could move funds, such as `stakeLockedCurveLp` and alike